wpilibNewCommands/src/main/native/cpp/frc2/command/Commands.cpp

```diff
@@ -73,15 +74,19 @@ CommandPtr cmd::Print(std::string_view msg) {
   return PrintCommand(msg).ToPtr();
 }
 
-WPI_IGNORE_DEPRECATED
 CommandPtr cmd::DeferredProxy(wpi::unique_function<Command*()> supplier) {
-  return ProxyCommand(std::move(supplier)).ToPtr();
+  return Defer(
+      [supplier = std::move(supplier)]() mutable {
+        return ProxyCommand{std::move(supplier())}.ToPtr();
```

Comment:
I couldn't make this work using the `AsProxy` decorator

Reply:
Since the return of `supplier()` is already an rvalue, the move isn't needed.
The following should work:
```cpp
return supplier().AsProxy()
```
(You can also pop a `ToPtr` there in between, but I think it's unneeded)

Reply:
Oh I missed that we're dealing with a `Command*` and not a `CommandPtr`.
Never mind.

Reply:
I don't think there's a non-owning version of `AsProxy()`, so yes, I think this does have to be via the constructor. Maybe just add a comment explaining that so we don't forget?

Reply:
The std::move is still unnecessary as the return value of `supplier` is `Command*`

Reply:
It is

Reply:
How so? I compiled locally with the following patch to 3ec23131c40db786dfdee96d629bd9a8928e897c:
```diff
diff --git a/wpilibNewCommands/src/main/native/cpp/frc2/command/Commands.cpp b/wpilibNewCommands/src/main/native/cpp/frc2/command/Commands.cpp
index 219d81537..84c056f1f 100644
--- a/wpilibNewCommands/src/main/native/cpp/frc2/command/Commands.cpp
+++ b/wpilibNewCommands/src/main/native/cpp/frc2/command/Commands.cpp
@@ -77,7 +77,9 @@ CommandPtr cmd::Print(std::string_view msg) {
 CommandPtr cmd::DeferredProxy(wpi::unique_function<Command*()> supplier) {
   return Defer(
       [supplier = std::move(supplier)]() mutable {
-        return ProxyCommand{std::move(supplier())}.ToPtr();
+        // There is no non-owning version of AsProxy(), so use the non-owning
+        // ProxyCommand constructor instead.
+        return ProxyCommand{supplier()}.ToPtr();
       },
       {});
 }
 ```

Reply:
whoops I read that message wrong :sweat_smile: